src/topology/flush_control.rs

```diff
@@ -0,0 +1,274 @@
+use crate::bounded_channel::{BoundedReceiver, BoundedSender, bounded};
+use std::sync::{Arc, Mutex};
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::{Receiver, Sender};
+use tower::BoxError;
+use tracing::warn;
+
+const FLUSH_CHAN_SIZE: usize = 20;
```

Comment:
What will happen to callers of this if the channel fills up for some reason?

Reply:
For senders it will block, which for the broadcast side in the lambda extension I'm wrapping it with a timeout method. I may need to look at the response channel to see if we need a timeout there too.